api.js

```diff
@@ -378,9 +378,9 @@ Api.prototype.syncStart = function (req, res, m, q) {
   if (!events) return onerror(res, 'Target not found')
 
   var debounceProgress = debounce(onprogress, q.interval || 2000)
-  events.on('progress', debounceProgress)
-  events.on('error', onend)
-  events.on('end', onend)
+  events.once('progress', debounceProgress)
```

Comment:
I think `once` is ok for the other events, but progress will definitely fire many times.